velox/functions/sparksql/tests/ArithmeticTest.cpp

```diff
@@ -268,6 +268,21 @@ TEST_F(ArithmeticTest, cosh) {
   EXPECT_TRUE(std::isnan(cosh(kNan).value_or(0)));
 }
 
+TEST_F(ArithmeticTest, unhex) {
+  const auto unhex = [&](std::optional<std::string> a) {
+    return evaluateOnce<std::string>("unhex(c0)", a);
+  };
+
+  EXPECT_EQ(unhex("737472696E67"), "string");
+  EXPECT_EQ(unhex(""), "");
+  std::string b("#\0", 2);
+  EXPECT_EQ(unhex("123"), b);
```

Comment:
I'm confused about this tests case. I expect 1 to be converted to first byte and 23 to second bytes. 23 is a hex for # character, hence, I'd expect a strings with 2 bytes: first with ASCII code 1, second #.

\x01\x23

Reply:
Spark first process first byte, then process remaining bytes, when input is odd number length.

For this case `unhex("123")`, processing first byte, resultBuffer is `[1, 0]`, 
![image](https://github.com/facebookincubator/velox/assets/34979747/d7bfec14-5593-4a19-9dad-24041ff68432)

then processing remaining two bytes, resultBuffer is `[35, 0]`
![image](https://github.com/facebookincubator/velox/assets/34979747/b974ceca-4b45-4ecd-901c-9d45f02546a3)

So, result is two char, first is ASCII code #, second is ASCII code null.

Reply:
@Yohahaha This behavior is confusing to me. It looks like when number of bytes is odd, first byte is ignored, but a \0 is added to the end of the result. Would you double check this is indeed Spark's behavior? For example, what's the result when there is only 1 byte? Will result of 123 be same as n23, where 'n' is any digit? Will result of 123 is the same as as result of 23, except that it has a trailing null?

CC: @PHILO-HE @rui-mo

Reply:
```shell
spark-sql> select unhex("F"), length(unhex("F"));
	1

spark-sql> select unhex("b"), length(unhex("b"));

	1

spark-sql> select unhex("b23"), length(unhex("b23"));
#	2

spark-sql> select unhex("123"), length(unhex("123"));
#	2

spark-sql> select unhex("23"), length(unhex("23"));
#	1
```

Reply:
@rui-mo @PHILO-HE would you help double check this behavior? thanks!

Reply:
@Yohahaha I double-checked and found the behavior was aligned with Spark. Could you help add more tests, like unhex("b23"), unhex("23") and unhex("b2323") etc ?

Reply:
thanks, added.